Neos.ContentRepositoryRegistry/Classes/Command/CrCommandController.php

```diff
@@ -144,15 +144,38 @@ public function replayAllCommand(string $contentRepository = 'default', bool $qu
         }
     }
 
+    /**
+     * Resets all projections of the specified Content Repository
+     *
+     * @param string $contentRepository Identifier of the Content Repository instance to operate on
+     * @param bool $quiet If set only fatal errors are rendered to the output
+     */
+    public function resetAllCommand(string $contentRepository = 'default', bool $quiet = false): void
```

Comment:
I agree here. Actually I struggle to understand what `reset` really means. What does it do? Why and when should someone use this command?

TBH I would have expected a better description in the PR for that as it will get into the changelog.

Reply:
Yes this pr is not to be merged as is :)

my proposed new command will reset all the projections. This is implicit the first step of replayAll.

The reasoning to include is are: that we have a command to prune everything `cr:prune` and as the cr api allows to throw away all projections (like a cache flush) we should also allow this.

That command will most likely only be used when `cr:setup` fails because there is data in the projections. Basically for a super clean setup. Which hopefully is never need in 9.0 stable but was for example for https://github.com/neos/neos-development-collection/pull/4791.

Reply:
Then we probably should include this change in some way and update the PR description of #4791 afterwards 😉. As of now the description is misleading.

Reply:
the _real_ hopefully not that present usecase past 9.0 would be `cr:setup --reset-projections`.
Maybe that would be the way to go?

But i also prefer those commands:

```
cr:resetAllProjections

cr:replayAllProjections

cr:replayProjection
```

instead of the implicit `cr:replayAll`

Reply:
`cr:projectionReplayAll`

`cr:projectionReplay`

prefer `cr:setup --reset-projections` over `cr:projectionResetAll`